lib/ASRouterTargeting.jsm

```diff
@@ -337,6 +337,8 @@ this.ASRouterTargeting = {
       return false;
     } else if (!candidateMessageTrigger.params) {
       return true;
+    } else if (candidateMessageTrigger.patterns) {
```

Comment:
Shouldn't the logic here be

```js
else if (!candidateMessageTrigger.params && !candidateMessageTrigger.patterns) {
  return true;
}
```

Reply:
This is abusing the fact that we have only 1 message with patterns. I needed a condition to exit early and with `true` because for the next check we would essentially have to duplicate the `MatchPatternSet` checks from `TriggerListener.jsm`
I mentioned in the first comment that I would rather fix this as a follow up. Or come up with a better idea that creating a new `MatchPatternSet`.

Reply:
Got it, I think this might break pretty quickly if we start landing test messages / fallbacks; that said, happy if this just has a comment and a follow-up is filed with `P1`